examples/core-auth-nextjs-email/src/auth.ts

```diff
@@ -0,0 +1,99 @@
+import NextAuth from 'next-auth';
+import { AuthProvider, SupportedAuthProvider } from '@toolpad/core';
+import GitHub from 'next-auth/providers/github';
+// import Credentials from 'next-auth/providers/credentials';
+
+import Nodemailer from 'next-auth/providers/nodemailer';
+import { PrismaAdapter } from '@auth/prisma-adapter';
+import type { Provider } from 'next-auth/providers';
+import { prisma } from './prisma';
+
+const providers: Provider[] = [
+  GitHub({
+    clientId: process.env.GITHUB_CLIENT_ID,
+    clientSecret: process.env.GITHUB_CLIENT_SECRET,
+  }),
+  // Credentials({
+  //   credentials: {
+  //     email: { label: 'Email Address', type: 'email' },
+  //     password: { label: 'Password', type: 'password' },
+  //   },
+  //   authorize(c) {
+  //     if (c.password !== 'password') {
+  //       return null;
+  //     }
+  //     return {
+  //       id: 'test',
+  //       name: 'Test User',
+  //       email: String(c.email),
+  //     };
+  //   },
+  // }),
+  Nodemailer({
+    server: {
+      host: process.env.EMAIL_SERVER_HOST,
+      port: process.env.EMAIL_SERVER_PORT,
+      auth: {
+        user: process.env.EMAIL_SERVER_USER,
+        pass: process.env.EMAIL_SERVER_PASSWORD,
+      },
+      secure: true,
+    },
+    from: process.env.EMAIL_FROM,
+  }),
+];
+
+export const providerMap = providers.map((provider) => {
+  if (typeof provider === 'function') {
+    const providerData = provider();
+    return {
+      id: providerData.id as SupportedAuthProvider,
```

Comment:
This doesn't seem necessary here https://github.com/mui/toolpad/blob/4e00a5fc342babd9c900c24cc41359d5a3a985ba/playground/nextjs/src/auth.ts#L29-L35

```suggestion
      id: providerData.id,
```

Reply:
I believe that shouldn't be necessary either. At most I'd expect the export to be typed

```tsx
export const providerMap: AuthProvider[] = // ...
```